tasks/dnsmasq.yml

```
@@ -37,7 +37,7 @@
     mode: 0644
   become: true
   notify: restart dnsmasq
-  when: "{{ item.when }}"
+  when: item.when
```

```suggestion
  when: "{{ dnsmasq_item.when }}"
```

See #488.

---

@nre-ableton 
The lint error was about jinja expressions being used in `when`. It only expected pure python expressions. Have you tried this out by lint?

---

@sandipb True, but there are several other places in this role where such expressions are used, which is why my PR disabled the expression for the time being. I think that changing the other usages and re-enabling the warning deserves a separate PR.

However, my suggestion is not so much about the Jinja2 expressions but rather the use of `item`, which will be undefined and fail. That was named to `dnsmasq_item` [here](https://github.com/ansible-community/ansible-consul/commit/3cb15f3de7a6b29e37d55f90f52e446141af19d8#diff-6c7ae99af38f268480961efb10a100f3c9a521c6cbeb0f6cab760ad42d1f625aR45-R46).

If you prefer, you could also do:

```suggestion
  when: dnsmasq_item.when
```

---

Duplicate